### ISSUE TITLE
New compiler: ternary operator parsing troubles with nested expressions

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#2069]
CC @fernewelten

The new compiler fails to correctly parse the nested expressions inside the first "term" of a ternary operator.

Consider following script:
```
int a = 10;
int b = a < 5 ? (a + 1) : a; // passes
int c = a < 5 ? (a + (a + 1)) : a; // fails
int d = a < 5 ? a : (a + (a + 1)); // passes
```

Compiler will throw error on `c` assignment:
> Test.asc(15): Internal error: Unexpected ')' after 1st term of ternary

Also, note that it sais "internal error" for some reason.

**Comment by fernewelten** [comment]
I'll look into this.

**Comment by ericoporto** [comment]
Is this issue related to https://github.com/adventuregamestudio/ags/pull/2510 ?

**Comment by ivan-mogilko** [comment]
I forgot about tests, will add some.

**Comment by ivan-mogilko** [comment]
@ericoporto hmm, there's a potential issue with the tests which I did not realize,
"make_release" task is not run in CI if tests failed.
But some bugs may take long time to fix.

Is it possible to make releases still done if these tests fail? or particular tests excluded form this rule?
Not sure what's a better way to deal with this to both have an indication of a failure, and releases.

**Comment by fernewelten** [comment]
Okay, the whole test case ought to pass, and I'm looking at it right now. It seems that the expression parser doesn't correctly delimit the parts of the ternary.

**Comment by ivan-mogilko** [comment]
Fixed by #2523
